gnomad/utils/filtering.py

```diff
@@ -529,3 +529,57 @@ def split_vds_by_strata(
     )
 
     return [hl.vds.filter_samples(vds, list(s)) for strata, s in s_by_strata.items()]
+
+
+def remove_items_from_freq(
```

Comment:
I tried to return freq_meta_expr, then when I annotate it back, I got `source mismatch` issue, I discussed with Mike, he was doing similar to remove `downsampling`, he suggested to filter the freq_meta in place.

Reply:
Can you show me where you annotate it back? I can help fix that error rather than change the standard methodology we use for our gnomad_methods functions

Reply:
This was what I had at first: 
```
def remove_items_from_freq(ht: hl.Table, items_to_remove: Dict[str, str])-> hl.Table:
    """
    Script to remove items from the freq array and freq_meta array in the Table.

    :param ht: Input Table.
    :param items_to_remove: Dictionary of items to remove from the freq array and freq_meta array.
    :return: Table with specified items removed from the freq array and freq_meta array.
    """

    freq = hl.map(lambda x: x[0].annotate(meta=x[1]), hl.zip(ht.freq, ht.freq_meta))

    for k, v in items_to_remove.items():
        freq = hl.filter(lambda f: (~f.meta.contains(k) | (f.meta.get(k) != v)), freq,)
        freq_meta = hl.filter(lambda m: (~m.contains(k) | (m.get(k) != v)), ht.freq_meta,)
    
    ht = ht.annotate(freq=freq.map(lambda x: x[0:4]))
    ht = ht.annotate_globals(freq_meta=freq_meta)
    
    return ht

import hail as hl
from gnomad_qc.v3.resources.release import release_ht_path
from typing import Dict

ht = hl.read_table(release_ht_path(release_version="3.1.2"))

d = {'pop':'han'}
ht = remove_items_from_freq(ht, d)
```
error message: 
```
ExpressionException: 'Table.annotate_globals: field 'freq_meta'': source mismatch
  Expected an expression from source <hail.table.Table object at 0x7fc9a6d3ccd0>
  Found expression derived from source <hail.table.Table object at 0x7fc9a6d99a20>
  Problematic field(s): ['freq_meta']
```

Reply:
This was before I consider the case where I might have multiple pops to remove.

Reply:
```
def remove_items_from_freq(
    freq_expr: hl.expr.ArrayExpression,
    freq_meta_expr: hl.expr.ArrayExpression,
    items_to_remove: Union[Dict[str, List[Any]], List[Any]]
) -> hl.Table:
    """
    Script to remove items from the freq array and freq_meta array in the Table.

    :param ht: Input Table with freq and freq_meta arrays.
    :param items_to_remove: Dictionary or list of items to remove from the freq and freq_meta arrays.
    :return: Table with specified items removed from the freq array and freq_meta array.
    """
    freq_meta_expr = freq_meta_expr.collect(_localize=False)[0]
    def _remove_key_value_pair_from_freq(
        freq_expr: hl.expr.ArrayExpression,
        freq_meta_expr: hl.expr.ArrayExpression,
        key: str,
        value: str,
    ) -> hl.Table:
        """
        Remove key-value pair from freq and freq_meta arrays.

        :param key: Key to remove from freq_meta array.
        :param value: Value to remove from freq_meta array.
        :param ht: Input Table with freq and freq_meta arrays.
        :return: Table with specified key-value pair removed from freq and freq_meta arrays.
        """
        freq_expr = hl.map(lambda x: x[0].annotate(meta=x[1]), hl.zip(freq_expr, freq_meta_expr))

        freq_expr = hl.filter(
            lambda f: (~f.meta.contains(key) | (f.meta.get(key) != value)),
            freq_expr,
        )
        freq_expr = freq_expr.map(lambda x: x[0:4])
        freq_meta_expr = freq_meta_expr.filter(
                lambda m: ~m.contains(key) | (m.get(key) != value)
            )

        return freq_expr, freq_meta_expr

    if isinstance(items_to_remove, list):
        freq_expr = hl.map(lambda x: x[0].annotate(meta=x[1]), hl.zip(freq_expr, freq_meta_expr))
        for key in items_to_remove:
            freq_expr = hl.filter(lambda f: ~f.meta.contains(key), freq_expr)
        freq_expr = freq_expr.map(lambda x: x[0:4])
        freq_meta_expr = freq_meta_expr.filter(lambda m: ~m.contains(key))

    elif isinstance(items_to_remove, dict):
        for k, v in items_to_remove.items():
            for value in v:
                freq_expr, freq_meta_expr = _remove_key_value_pair_from_freq(
                    freq_expr, freq_meta_expr, k, value
                )

    return freq_expr, freq_meta_expr
```